presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -42,6 +43,17 @@ static void checkArrayRange(int[] array, int offset, int length)
         }
     }
 
+    static void checkArrayRange(boolean[] array, int offset, int length)
+    {
+        if (array == null) {
```

Comment:
the other checkArrayRange method on L38 doesn't allow null input; is there any particular reason to allow nulls here? Would it be possible to move null check to the caller and make this method throw on null input like the one one?

Reply:
The blocks may not always have the null array. When it doesn't have null, there is no need to have the isNull array in the block. I modified it to the following:
```
static void checkArrayRange(boolean[] array, int offset, int length)
    {
        requireNonNull(array, "array is null");
        if (offset < 0 || length < 0 || offset + length > array.length) {
            throw new IndexOutOfBoundsException(format("Invalid offset %s and length %s in array with %s elements", offset, length, array.length));
        }
    }

static boolean[] appendNullToIsNullArray(@Nullable boolean[] isNull, int offsetBase, int positionCount)
    {
        int desiredLength = offsetBase + positionCount + 1;
        boolean[] newIsNull = new boolean[desiredLength];
        if (isNull != null) {
            checkArrayRange(isNull, offsetBase, positionCount);
            System.arraycopy(isNull, 0, newIsNull, 0, desiredLength - 1);
        }
        newIsNull[desiredLength - 1] = true;
        return newIsNull;
    }
```